tests/compatibility/main_test.go

```diff
@@ -37,49 +46,89 @@ func testMain(m *testing.M) int {
 
 func setup() {
 	var err error
-	emulatorHost := fmt.Sprintf("%s:8085", os.Getenv("PUBSUB_TEST_HOST"))
+
+	env := os.Getenv("APP_ENV")
+	if env == "" {
+		env = "dev"
+	}
+
+	var appConfig map[string]interface{}
+	if err = configreader.NewDefaultConfig().Load(env, &appConfig); err != nil {
+		os.Exit(1)
+	}
+	if _, ok := appConfig["admin"]; ok {
+		adminUser = appConfig["admin"].(map[string]interface{})["username"].(string)
+		adminPassword = appConfig["admin"].(map[string]interface{})["password"].(string)
+	}
+
+	emulatorHost = fmt.Sprintf("%s:8085", os.Getenv("PUBSUB_TEST_HOST"))
+	metroHost = fmt.Sprintf("%s:8081", os.Getenv("METRO_TEST_HOST"))
+	metroHTTPHost = fmt.Sprintf("http://%s:8082", os.Getenv("METRO_TEST_HOST"))
+
+	createProjectInMetro()
+
 	emulatorClient, err = pubsub.NewClient(context.Background(), projectId,
 		option.WithEndpoint(emulatorHost),
 		option.WithoutAuthentication(),
 		option.WithGRPCDialOption(grpc.WithInsecure()),
 	)
 	if err != nil {
-		os.Exit(1)
+		os.Exit(2)
 	}
-	metroHost := fmt.Sprintf("%s:8081", os.Getenv("METRO_TEST_HOST"))
-	metroClient, err = pubsub.NewClient(context.Background(), projectId,
-		option.WithEndpoint(metroHost),
-		option.WithoutAuthentication(),
-		option.WithGRPCDialOption(grpc.WithInsecure()),
-	)
+
+	metroClient, err = metro_pubsub.NewMetroClient(metroHost, false, projectId, metro_pubsub.Credentials{Username: username, Password: password})
 	if err != nil {
-		os.Exit(2)
+		os.Exit(3)
 	}
-	// create project in metro
-	createProjectInMetro()
 }
 
 func createProjectInMetro() {
-	url := fmt.Sprintf("http://%s:8082/v1/projects", os.Getenv("METRO_TEST_HOST"))
+	url := fmt.Sprintf("%s/v1/projects", metroHTTPHost)
 	payload := bytes.NewBuffer([]byte("{\"name\": \"" + projectId + "\",\"projectId\": \"" + projectId + "\"}"))
-	r, err := http.Post(url, "application/json", payload)
+	req, err := http.NewRequest(http.MethodPost, url, payload)
+	if err != nil {
+		os.Exit(4)
+	}
+	req.SetBasicAuth(adminUser, adminPassword)
+	r, err := http.DefaultClient.Do(req)
 	if err != nil || r.StatusCode != 200 {
-		os.Exit(3)
+		os.Exit(5)
+	}
+	setupProjectCredentials()
+}
+
+func setupProjectCredentials() {
+	var parsedResponse map[string]string
+
+	url := fmt.Sprintf("%s/v1/projects/%s/credentials", metroHTTPHost, projectId)
+	payload := bytes.NewBuffer([]byte(fmt.Sprintf(`{"username": "%s", "password":"password"}`, projectId+"_user")))
```

Comment:
do we require any payload for this, i thought just POST on the url creates new creds

Reply:
yes
```
rpc CreateProjectCredentials(ProjectCredentials) returns (ProjectCredentials){
      option (google.api.http) = {
          post: "/v1/projects/{project_id}/credentials"
          body: "*"
      };
      option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
          summary: "Create new project credentials for an app"
          description: "Credentials to be used for all operations under this project by the app"
          tags: "Project"
      };
  }
```